src/Quartz/Simpl/SimpleJobFactory.cs

```diff
@@ -88,7 +88,15 @@ public virtual IJob NewJob(TriggerFiredBundle bundle, IScheduler scheduler)
     /// </summary>
     public virtual void ReturnJob(IJob job)
     {
-        var disposable = job as IDisposable;
-        disposable?.Dispose();
+        if (job is IDisposable disposableJob)
+        {
+            disposableJob.Dispose();
+        }
+        // check for wrapped jobs only if the current job is not disposable
+        // disposable wrappers should handle inner disposal on it's own
```

Comment:
Correct Disposing via ReturnJob is not really a trivial thing. For things like MicrosoftDI we need to have in mind, that we should dispose only if the scope does not handle the disposal on it's own.

I think we can assume, that the first disposable Job(-Wrapper) should be responsible for disposal down the road, so we just search for the first IDisposable if there is any and then we stop checking.